bindings/rust/evmc-vm/src/lib.rs

```diff
@@ -61,6 +68,223 @@ impl ExecutionResult {
     }
 }
 
+impl<'a> ExecutionContext<'a> {
+    pub fn new(
+        revision: &'a ffi::evmc_revision,
+        message: &'a ffi::evmc_message,
+        context: &'a mut ffi::evmc_context,
+        instance: &'a mut ffi::evmc_instance,
+    ) -> Self {
+        ExecutionContext {
+            rev: revision,
+            msg: message,
+            ctx: context,
+            inst: instance,
+        }
+    }
+
+    pub fn revision(&self) -> &ffi::evmc_revision {
+        &self.rev
+    }
+
+    pub fn call_kind(&self) -> &ffi::evmc_call_kind {
+        &self.msg.kind
+    }
+
+    pub fn is_static(&self) -> bool {
+        self.msg.flags == 1
+    }
+
+    pub fn flags(&self) -> u32 {
+        self.msg.flags
+    }
+
+    pub fn calldepth(&self) -> i32 {
+        self.msg.depth
+    }
+
+    pub fn gas(&self) -> i64 {
+        self.msg.gas
+    }
+
+    pub fn destination(&self) -> &ffi::evmc_address {
+        &self.msg.destination
+    }
+
+    pub fn sender(&self) -> &ffi::evmc_address {
+        &self.msg.sender
+    }
+
+    pub fn input_data(&self) -> Option<Vec<u8>> {
+        if !self.msg.input_data.is_null() {
+            let mut ret: Vec<u8> = Vec::with_capacity(self.msg.input_size);
+            unsafe {
+                std::ptr::copy(self.msg.input_data, ret.as_mut_ptr(), self.msg.input_size);
+            }
+            Some(ret)
+        } else {
+            None
+        }
+    }
+
+    pub fn value(&self) -> &ffi::evmc_uint256be {
+        &self.msg.value
+    }
+
+    pub fn create2_salt(&self) -> &ffi::evmc_bytes32 {
+        &self.msg.create2_salt
+    }
+
+    pub fn account_exists(&mut self, addr: &ffi::evmc_address) -> bool {
+        unsafe {
+            assert!(!(*self.ctx.host).account_exists.is_some());
```

Comment:
I think the assertions should take place in `new`.

However, in the long-run, no such assertions should be made, just catch panics in the macro and return an EVMC failure.

Reply:
this makes sense.